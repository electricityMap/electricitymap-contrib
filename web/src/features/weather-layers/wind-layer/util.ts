/**
 * @returns {Boolean} true if the specified value is not null and not undefined.
 */
export function isValue(x: unknown) {
  return x !== null && x !== undefined;
}

/**
 * @returns {Number} returns remainder of floored division, i.e., floor(a / n). Useful for consistent modulo
 * of negative numbers. See http://en.wikipedia.org/wiki/Modulo_operation.
 */
export function floorModulus(a: number, n: number) {
  return a - n * Math.floor(a / n);
}
