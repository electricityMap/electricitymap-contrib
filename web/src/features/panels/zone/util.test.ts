import { showEstimationFeedbackCard } from './util';

describe('showEstimationFeedbackCard', () => {
  beforeEach(() => {
    localStorage.clear();
  });

  it('should return true when feedbackCard is currently not shown, card has not been seen before and it has been collapsed', () => {
    const result = showEstimationFeedbackCard(1, false, 'notSeenBefore');
    expect(result).to.equal(true);
  });

  it('should return true when showFeedbackCard is true', () => {
    const result = showEstimationFeedbackCard(1, true, 'seenBefore');
    expect(result).to.equal(true);
  });

  it('should return false when conditions are not met', () => {
    const result = showEstimationFeedbackCard(0, false, 'seenBefore');
    expect(result).to.equal(false);
  });

  it('should return false when conditions are not met', () => {
    const result = showEstimationFeedbackCard(2, false, 'seenBefore');
    expect(result).to.equal(false);
  });
});
