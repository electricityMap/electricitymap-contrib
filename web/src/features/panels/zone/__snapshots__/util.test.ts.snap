// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getContributors > should return contributors for a zone 1`] = `
{
  "contributors": [
    "Manu1400",
    "autipial",
    "q--",
    "florianscheidl",
    "MisterClean",
    "corradio",
    "nessie2013",
    "tmslaine",
    "systemcatch",
    "alixunderplatz",
    "brandongalbraith",
    "jarek",
    "unitrium",
    "amv213",
    "chuqtas",
    "ajoga",
    "MariusKroon",
    "pascalheraud",
    "VIKTORVAV99",
    "MuffinCompiler",
    "Fred-3000",
    "gadakast",
    "yurukov",
    "ultibo",
    "bogomil",
    "Pantkowsky",
    "MindFreeze",
    "igorarduin",
    "skovhus",
    "alexanmtz",
    "admetny",
    "mlucchini",
    "daveachuk",
    "felixvanoost",
    "lav-julien",
    "scriptator",
    "misterclean",
    "christopheradlam",
    "reinvented",
    "pjakobsen",
    "madsnedergaard",
    "KabelWlan",
    "postronium",
    "Kikof2",
    "veqtrus",
    "Frantisek12",
    "bohne13",
    "chiefymuc",
    "IV1T3",
    "sin",
    "hectorespert",
    "JosepRey",
    "pnook",
    "PETILLON-Sebastien",
    "lorrieq",
    "brunolajoie",
    "msaynevirta",
    "maxbellec",
    "AlexandreCouderc",
    "kruschk",
    "PaulCornelissen",
    "RolandSzep",
    "yuvalabou",
    "gopikrishna1793",
    "willbeaufoy",
    "GitMatze",
    "pierresegonne",
    "sahitya-pavurala",
    "mathilde-daugy",
    "byronwilliams",
    "SNPerkin",
    "wobniarin",
    "PPsyrius",
    "kongkille",
    "clawfire",
    "lgrigoryeva1",
    "Impelon",
    "silimotion",
    "Subarna578",
    "BobGrun",
    "davidcole1340",
    "drc38",
    "paulotaviodr",
    "7hibault",
    "r24mille",
    "xomanuel",
    "wojciej",
    "filipc",
    "Musketeiro",
    "RGarrido03",
    "mzaharie",
    "denplis",
    "JeanBaptisteScellier",
    "aekrylov",
    "augustekman",
    "RRyyas",
    "dpkruczek",
    "olivierSemet",
    "amkozlov",
    "wrma6",
    "robertahunt",
    "kepiej",
    "danielmatsuda",
    "miniksa",
    "ZachEichen",
    "ananyasharma20",
  ],
  "zoneContributorsIndexArray": [
    5,
    6,
    21,
    52,
  ],
}
`;

exports[`getDisclaimer > should return disclaimer for a zone 1`] = `"Solar production is collected from NED.nl and can contain their own estimations."`;
